src/plugins/unified_field_list/public/components/field_stats/field_stats.tsx

```diff
@@ -240,39 +234,86 @@ const FieldStatsComponent: React.FC<FieldStatsProps> = ({
-          })}
-        </EuiText>
-      </>
+  function combineWithTitleAndFooter(el: React.ReactElement) {
+    const countsElement = totalDocuments ? (
+      <EuiText color="subdued" size="xs" data-test-subj={`${dataTestSubject}-statsFooter`}>
+        {sampledDocuments && sampledDocuments < totalDocuments ? (
+          <FormattedMessage
+            id="unifiedFieldList.fieldStats.calculatedFromSampleRecordsLabel"
+            defaultMessage="Calculated from {sampledDocumentsFormatted} sample {sampledDocuments, plural, one {record} other {records}}."
+            values={{
+              sampledDocuments,
+              sampledDocumentsFormatted: (
+                <strong>
+                  {fieldFormats
+                    .getDefaultInstance(KBN_FIELD_TYPES.NUMBER, [ES_FIELD_TYPES.INTEGER])
+                    .convert(sampledDocuments)}
+                </strong>
+              ),
+            }}
+          />
+        ) : (
+          <FormattedMessage
+            id="unifiedFieldList.fieldStats.calculatedFromTotalRecordsLabel"
+            defaultMessage="Calculated from {totalDocumentsFormatted} {totalDocuments, plural, one {record} other {records}}."
+            values={{
+              totalDocuments,
+              totalDocumentsFormatted: (
+                <strong>
+                  {fieldFormats
+                    .getDefaultInstance(KBN_FIELD_TYPES.NUMBER, [ES_FIELD_TYPES.INTEGER])
+                    .convert(totalDocuments)}
+                </strong>
+              ),
+            }}
+          />
+        )}
+      </EuiText>
+    ) : (
+      <></>
     );
-  }
 
-  if (field.type === 'murmur3') {
     return (
       <>
-        <EuiText size="s">
-          {i18n.translate('unifiedFieldList.fieldStats.notAvailableForMurmur3FieldDescription', {
-            defaultMessage: `Summary information is not available for murmur3 fields.`,
-          })}
-        </EuiText>
+        {title ? <div data-test-subj={`${dataTestSubject}-title`}>{title}</div> : <></>}
+
+        <EuiSpacer size="s" />
+
+        {el}
+
+        {overrideFooter ? (
+          overrideFooter?.({ element: countsElement, totalDocuments, sampledDocuments })
+        ) : (
+          <>
+            <EuiSpacer size="m" />
+            {countsElement}
+          </>
+        )}
       </>
     );
   }
 
-  if (field.type === 'geo_point' || field.type === 'geo_shape') {
-    return overrideMissingContent ? overrideMissingContent() : null;
-  }
-
   if (
     (!histogram || histogram.buckets.length === 0) &&
     (!topValues || topValues.buckets.length === 0)
   ) {
-    return overrideMissingContent ? overrideMissingContent({ noDataFound: true }) : null;
+    const message = (
+      <FieldSummaryMessage
+        message={i18n.translate(
+          'unifiedFieldList.fieldStats.notAvailableForThisFieldWithoutDataDescription',
+          {
+            defaultMessage: `Analysis is not available for this field.`,
```

Comment:
@ninoslavmiskovic @gchaps Shall we have a separate message for when data is not available? Currently it also shows "Analysis is not available for this field" but we were also discussing bringing "This field is not available for visualizations because it doesn't have any data." copy. I think it would make sense to extend the copy then with "... for this search criteria" or "... based on sample records" but the copy would become too long. Wdyt?

Reply:
Perhaps it is implicit, and we can shorten the copy to just say: "This field does not contain any data" ? @gchaps - Wdyt?

Reply:
@ninoslavmiskovic It can happen that the field has some data but those documents were not a part of the sample. So it would be great to mention why no data: either it's not a part of the sample or it's not present for the current search query/filters.

Reply:
@jughosta Will something like this work?

> No field data for the current search.
> No field data for the current sample.

Reply:
@gchaps Thanks, that works!

Would it be okay to extend the second line to `No field data for the current sample of X records.`

Reply:
I think maybe we can remove the word current and shorten it a bit:

> No field data for X sample records.